syntax = "proto3";

package depthrender;

// The greeting service definition.
service DepthRender {
  rpc SetCameraParams (SetCameraParamsRequest) returns (SetCameraParamsReply) {}
  rpc RenderDepthImage (RenderDepthImageRequest) returns (RenderDepthImageReply) {}
}

message SetCameraParamsRequest{
  int32 width = 1;
  int32 height = 2;
  double fx = 3;
  double fy = 4;
  double cx = 5;
  double cy = 6;
}

message SetCameraParamsReply {
}

message RenderDepthImageRequest{
  int32 num_vertices = 1;
  int32 num_triangles = 2;
  bytes vertices = 3;
  bytes triangles = 4;
}

message RenderDepthImageReply{
  bytes depth_image = 1;
}
